lib/xcodeproj/project/project_helper.rb

```diff
@@ -157,6 +157,15 @@ def self.configuration_list(project, platform, deployment_target = nil, target_p
 
         cl.build_configurations << release_conf
         cl.build_configurations << debug_conf
+
+        project.build_configurations.each do |configuration|
+          next if %w(Debug Release).include?(configuration.name)
```

Comment:
So, take #2 (because I really hate array.append when map will suffice):

```ruby
cl.build_configurations  = project.build_configurations.map do |configuration|
  config = project.new(XCBuildConfiguration)
  config.name = configuration.name
  if config is debug or release
    config.build_settings = common_build_settings(configuration.name.downcase.to_sym, platform, deployment_target, target_product_type)
  end
  config
end
```

This removes 8+ so lines above too :wink: